react/features/base/connection/actions.native.ts

```diff
@@ -16,12 +19,33 @@ export * from './actions.any';
  * @returns {Function}
  */
 export function connect(id?: string, password?: string) {
-    return (dispatch: IStore['dispatch']) => dispatch(_connectInternal(id, password))
+    return (dispatch: IStore['dispatch'], getState: IStore['getState']) => {
+        const state = getState();
+        const { jwt } = state['features/base/jwt'];
+        const { iAmRecorder, iAmSipGateway } = state['features/base/config'];
+
+        if (!iAmRecorder && !iAmSipGateway && isVpaasMeeting(state)) {
+            return dispatch(getCustomerDetails())
+                .then(() => {
+                    if (!jwt) {
+                        return getJaasJWT(state);
+                    }
+                })
+                .then(j => {
+                    j && dispatch(setJWT(j));
+
+                    return dispatch(_connectInternal(id, password));
+                });
+        }
+
+        dispatch(_connectInternal(id, password))
+
         .catch(error => {
```

Comment:
Don't we need this catch in the other call to connectInternal?

Reply:
Why would we need it there? This error is related to visitors, not with guests or authenticated participants.

Reply:
Ah, good point!